wire-library/wire-runtime-swift/src/main/swift/JSONString.swift

```diff
@@ -66,8 +68,31 @@ public struct JSONString<T : Hashable> : Hashable, Codable {
                 try container.encode(String(value))
             }
         default:
-            fatalError("Unsupported type \(T.self)")
+            if T.self is Optional<Int64>.Type || T.self is Optional<UInt64>.Type {
```

Comment:
nit: It's weird to have this in the default case. Can we add these as regular cases above, like:
```
case Optional<Int64>.none, Optional<UInt64>.none:
   var container = encoder.singleValueContainer()
   try container.encodeNil()
default:
  fatalError("Unsupported type \(T.self)")
```

Reply:
Unforunately, this was the best place I could think to place this code. We cannot cast `T` as an optional due to the following error: `Cannot downcast from 'T' to a more optional type 'Optional<Int64>'`.

The logic flow is check if there is a non-nil value, then check the type to see if it is optional of the allowed int values. Otherwise, the type is not supported.